src/test/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpFilterTest.java

```diff
@@ -1,17 +1,78 @@
 package se.liu.ida.hefquin.engine.queryplan.executable.impl.ops;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Iterator;
+
+import org.apache.jena.datatypes.xsd.XSDDatatype;
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import org.apache.jena.sparql.expr.Expr;
+import org.apache.jena.sparql.util.ExprUtils;
 import org.junit.Test;
 
+import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.data.utils.SolutionMappingUtils;
+import se.liu.ida.hefquin.engine.queryplan.executable.ExecOpExecutionException;
+import se.liu.ida.hefquin.engine.queryplan.executable.IntermediateResultElementSink;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.GenericIntermediateResultBlockImpl;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.pullbased.TestUtils;
+
 public class ExecOpFilterTest
 {
 	@Test
-	public void test1_RenameTheseTestMethodsToSomethingThatDescribesTheirPurpose() {
+	public void filter_Numbers() {
+		final FilterTestSink sink = new FilterTestSink();
```

Comment:
Instead of creating your own `FilterTestSink`, simply use a `MaterializingIntermediateResultElementSink` here. That one materializes all the solution mappings that are sent to it into an internal data structure to which you can get access via the `getMaterializedIntermediateResult()` function of this class.